tests/console/snapper_cleanup_timeline.pm

```diff
@@ -0,0 +1,61 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Configure snapper and verify that timeline cleanup algorithm behaves accordingly.
+# Maintainer: QA SLE yast team <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub pre_run_hook {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+    record_info("Configuration", "Configure snapper.");
+    foreach my $param (keys %{$test_data->{snapper_config}}) {
+        assert_script_run("snapper set-config $param=$test_data->{snapper_config}->{$param}",
+            fail_message => "Snapper configuration failed for parameter $param");
+    }
+}
+
+sub run {
+    record_info("Check quota", "Verify that the percentage of root filesystem quota is less than 50%");
+    my $convert_numeric = qr/[^0-9.]+/;
+    my $disk_space      = script_output("df -h /  | awk \'NR==2 {print \$4}\'");
+    $disk_space =~ s/$convert_numeric//;
+    my $qgroup_space = script_output("btrfs qgroup show -F --sync --si / | awk \'NR==3 {print \$2}\'");
+    $qgroup_space =~ s/$convert_numeric//;
+    die "Root filesystem quota take more than 50% of root disk space" if ($qgroup_space >= 0.5 * $disk_space);
```

Comment:
0.5 should be replaced by SPACE_LIMIT from test_data. Could you do that?

Reply:
same apply for the comments in those `record_info`s. `0.2` is also test_data, so those 20% and 80% should not be fixed, otherwise the test would be inconsistent if data is changed.

Reply:
done

Reply:
in the blog is referring to the filesystem in general not just size of that folder in memory, also in btrfs the size do not correspond exactly with it shows. I'm not expert on snapper but I would play safe using the information in the ticket or in the blog. I also didn't see that you use `btrfs qgroup show /`

Reply:
SPACE_LIMIT 0.5 is interpreted as  If you have 1TB hard drive, this means 500GB will be used for snapshots, so it is the fraction of the filesystems space the snapshots may use, so when you have 49% the algorithm perform its task.

Reply:
So, indeed the exclusive space of btrfs quota of group with id 1/0 should be maximum 50% of the total disk space. SPACE_LIMIT is by default 0.5, therefore in documentation it says "..since the snapshots need less than 50% of the space..".

Reply:
BTW, instead of checking if SPACE_LIMIT and FREE_LIMIT have the default values, I found it simpler to set them with the rest of the parameters. WDYT?

Reply:
yes, it is fine, unless you would like to test the default, which then we would need to read them first

Reply:
The alternative idea would be to check the current value of each parameter and if different from the one in test_data, change it, but just seems adding steps to the algorithm, instead of  reducing.